client/js/heat.js

```diff
@@ -0,0 +1,91 @@
+class HeatMapRow {
+  /**
+   * format for a row for heat map values obj
+   * @param {number[]} data binary arr of days where music was listened to
+   */
+  constructor(data) {
+    this.data = data;
+  }
+}
+
+// TODO: retrieve data from the backend instead of using a hardcoded value
+const TOTAL_LIKED = 13;
+// list of day indices where music was played
+const LIKED_MUSIC_HIST = [1, 4, 7];
+
+// ith value of 1 means ith latest video is music
+const likedMusicBinaryHist = makeBinaryArr(LIKED_MUSIC_HIST, TOTAL_LIKED);
+
+/**
+ * converts an array to binary arr
+ * @param {number[]} arr values will be indices that hold 1 in output arr
+ * @param {number[]} total size of output arr
+ * @returns binary arr of size total where ith value 1
+ * corresponds to value i in input arr
```

Comment:
This comment is hard to follow and is essentially just verbalizing the code rather than explaining it. Say something like `Array of 1's and 0's where all indices provided in "arr" are set to 1`